autoblocks/_impl/configs/config.py

```diff
@@ -0,0 +1,246 @@
+from autoblocks._impl.util import AnyTask
+from autoblocks._impl.util import AutoblocksEnvVar
+from autoblocks._impl.util import get_running_loop
+
+log = logging.getLogger(__name__)
+
+AutoblocksConfigValueType = TypeVar("AutoblocksConfigValueType")
+
+
+def is_testing_context() -> bool:
+    """
+    Note that we check for the presence of the CLI environment
+    variable and not the test case contextvars because the
+    contextvars aren't set until run_test_suite is called,
+    whereas a prompt manager might have already been imported
+    and initialized by the time run_test_suite is called.
+    """
+    return bool(AutoblocksEnvVar.CLI_SERVER_ADDRESS.get())
+
+
+def config_revisions_map() -> dict[str, str]:
+    """
+    The AUTOBLOCKS_CONFIG_REVISIONS environment variable is a JSON-stringified
+    map of config IDs to revision IDs. This is set in CI test runs triggered
+    from the UI.
+    """
+    if not is_testing_context():
+        return {}
+
+    config_revisions_raw = AutoblocksEnvVar.CONFIG_REVISIONS.get()
+    if not config_revisions_raw:
+        return {}
+    return json.loads(config_revisions_raw)  # type: ignore
+
+
+def make_request_url(config: RemoteConfig) -> str:
+    config_id = config.id
+    base = f"{API_ENDPOINT}/configs/{config_id}"
+    revision_id = config_revisions_map().get(config_id)
+    if revision_id and is_testing_context():
+        return f"{base}/revisions/{revision_id}"
+    if isinstance(config, LatestRemoteConfig):
+        return f"{base}/versions/latest"
+    elif isinstance(config, RemoteConfigWithVersion):
+        return f"{base}/versions/{config.version}"
+    elif isinstance(config, DangerouslyUseUndeployedRemoteConfig):
+        if isinstance(config.dangerously_use_undeployed, DangerouslyUseUndeployedWithRevision):
+            return f"{base}/revisions/{config.dangerously_use_undeployed.revision_id}"
+        else:
+            return f"{base}/revisions/latest"
+
+    raise ValueError(f"Invalid config type: {config}")
+
+
+def is_remote_config_refreshable(config: RemoteConfig) -> bool:
+    """
+    A config is refreshable if the user has specified that they want the latest deployed config
+    or the latest undeployed config.
+    """
+    return isinstance(config, LatestRemoteConfig) or (
+        isinstance(config, DangerouslyUseUndeployedRemoteConfig)
+        and isinstance(config.dangerously_use_undeployed, LatestDangerouslyUseUndeployed)
+    )
+
+
+async def get_remote_config(config: RemoteConfig, timeout: timedelta, api_key: str) -> RemoteConfigResponse:
+    resp = await global_state.http_client().get(
+        make_request_url(config=config),
+        timeout=timeout.total_seconds(),
+        headers={"Authorization": f"Bearer {api_key}"},
+    )
+    resp.raise_for_status()
+    resp_json = resp.json()
+    return RemoteConfigResponse(
+        id=resp_json.get("id"),
+        version=resp_json.get("version"),
+        value=resp_json.get("value"),
```

Comment:
should we access directly here instead of using get? might be preferable to have a keyerror here rather than let it continue in an invalid state?

Reply:
ya makes sense